airbyte-ci/connectors/live-tests/src/live_tests/commons/connection_objects_retrieval.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import json
+import logging
+from pathlib import Path
+from typing import Dict, Optional, Set
+
+from connection_retriever import ConnectionObject, retrieve_objects  # type: ignore
+
+from .models import Command, ConfiguredAirbyteCatalog, ConnectionObjects, SecretDict
+
+LOGGER = logging.getLogger(__name__)
+
+
+def parse_config(config: Dict | str | None) -> Optional[SecretDict]:
+    if not config:
+        return None
+    if isinstance(config, str):
+        return SecretDict(json.loads(config))
+    else:
+        return SecretDict(config)
+
+
+def parse_catalog(catalog: Dict | str | None) -> Optional[ConfiguredAirbyteCatalog]:
+    if not catalog:
+        return None
+    if isinstance(catalog, str):
+        return ConfiguredAirbyteCatalog.parse_obj(json.loads(catalog))
+    else:
+        return ConfiguredAirbyteCatalog.parse_obj(catalog)
+
+
+def parse_state(state: Dict | str | None) -> Optional[Dict]:
+    if not state:
+        return None
+    if isinstance(state, str):
+        return json.loads(state)
+    else:
+        return state
+
+
+CONNECTION_OBJECT_TYPE_TO_PARSER = {
+    ConnectionObject.SOURCE_CONFIG: parse_config,
+    ConnectionObject.CONFIGURED_CATALOG: parse_catalog,
+    ConnectionObject.STATE: parse_state,
+}
+
+
+def get_connector_config_from_path(config_path: Path) -> Optional[SecretDict]:
+    return parse_config(config_path.read_text())
+
+
+def get_state_from_path(state_path: Path) -> Optional[Dict]:
+    return parse_state(state_path.read_text())
+
+
+def get_catalog_from_path(path: Path) -> Optional[ConfiguredAirbyteCatalog]:
+    return parse_catalog(path.read_text())
+
```

Comment:
totally optional: Could we put this up into the `parse_` methods with this behavior if the type is `path`? Is this the only case in which our `parse_` methods would expect to receive strings as params?

Reply:
Do you mean `parse_` should accept an Union of Path or String and call read_text if it's a Path?
I'd prefer to keep separate functions as it might complexify the MyPy validation...
Both `get_*` and `parse_*` are directly used in  [`get_connection_objects`](https://github.com/airbytehq/airbyte/blob/f3f81124ee8a8cfa7455c2d72847b6950a4028c3/airbyte-ci/connectors/live-tests/src/live_tests/commons/connection_objects_retrieval.py#L74):
* `parse_` when we retrieve dicts from the db
* `get_` when we retrieve local paths from the CLI